File: Product-of-the-Last-K-Numbers.js
var ProductOfNumbers = function() 
{
    this.nums = [1];
};

ProductOfNumbers.prototype.add = function(n) 
{
    if (n === 0) 
    {
        this.nums = [1];
    } 
    else 
    {
        this.nums.push(this.nums[this.nums.length - 1] * n);
    }
};

ProductOfNumbers.prototype.getProduct = function(k) 
{
    if (k >= this.nums.length) return 0;
    return this.nums[this.nums.length - 1] / this.nums[this.nums.length - k - 1];
};

File: Product-of-the-Last-K-Numbers.java
class ProductOfNumbers 
{
    private List<Integer> nums;
    
    public ProductOfNumbers() 
    {
        nums = new ArrayList<>();
        nums.add(1);
    }
    
    public void add(int n) 
    {
        if (n == 0) 
        {
            nums.clear();
            nums.add(1);
        } 
        else 
        {
            nums.add(nums.get(nums.size() - 1) * n);
        }
    }
    
    public int getProduct(int k) 
    {
        if (k >= nums.size()) return 0;
        return nums.get(nums.size() - 1) / nums.get(nums.size() - k - 1);
    }
}
